cl/recap/tasks.py

```
@@ -2300,6 +2329,89 @@ def fetch_docket_by_pacer_case_id(session, court_id, pacer_case_id, fq):
+
+    # Check court connectivity, if fails retry the task, hopefully, it'll be
+    # retried in a different not blocked node
+    if not is_pacer_court_accessible(court_id):
+        if self.request.retries == self.max_retries:
+            msg = f"Blocked by court: {court_id}"
+            mark_fq_status(fq, msg, PROCESSING_STATUS.FAILED)
+            self.request.chain = None
+            return None
+        raise self.retry()
+
+    async_to_sync(mark_pq_status)(fq, "", PROCESSING_STATUS.IN_PROGRESS)
+
+    session_data = get_pacer_cookie_from_cache(fq.user_id)
+    if session_data is None:
+        msg = f"Cookie cache expired before task could run for user: {fq.user_id}"
+        mark_fq_status(fq, msg, PROCESSING_STATUS.FAILED)
+        self.request.chain = None
+        return None
+
+    s = ProxyPacerSession(
+        cookies=session_data.cookies, proxy=session_data.proxy_address
+    )
+
+    docket_number = fq.docket_number or getattr(
+        fq.docket, "docket_number", None
+    )
+    start_time = now()
+    try:
+        result = get_appellate_docket_by_docket_number(
+            docket_number=docket_number,
+            court_id=map_cl_to_pacer_id(court_id),
+            session_data=s,
+            **get_fq_appellate_docket_kwargs(fq),
+        )
+    except (requests.RequestException, ReadTimeoutError) as exc:
+        msg = "Network error while purchasing docket for fq: %s."
+        if self.request.retries == self.max_retries:
+            mark_fq_status(fq, msg, PROCESSING_STATUS.FAILED)
+            self.request.chain = None
+            return None
+        mark_fq_status(
+            fq, f"{msg}Retrying.", PROCESSING_STATUS.QUEUED_FOR_RETRY
+        )
+        raise self.retry(exc=exc)
+    except ParsingException:
+        msg = "Unable to purchase docket for fq: %s."
+        mark_fq_status(fq, msg, PROCESSING_STATUS.FAILED)
+        self.request.chain = None
+        return None
+
+    content_updated = result["content_updated"]
```

I noticed that `get_appellate_docket_by_docket_number` can return `None` if the parsed docket contains empty data. Could we mark the FQ as failed and add a related error message if this situation occurs?

Also, within `get_appellate_docket_by_docket_number`, `RequestException` is being caught, and if this happens, it returns `None`. Since the error is not propagated, `fetch_appellate_docket` won't be able to catch this error and mark it as failed with the message: `Network error while purchasing docket for fq: %s`. Perhaps it would be better to tweak the base logic of `get_appellate_docket_by_docket_number` to propagate the error.